internal/renderer/service_util.go

```diff
@@ -341,6 +343,51 @@ func createLbService4Gateway(c *RenderContext, gw *gwapiv1a2.Gateway) *corev1.Se
 		})
 	}
 
+	// Set health check port
+	regexProtocol := `^service\.beta\.kubernetes\.io\/.*health.*protocol$`
+	regexPort := `^service\.beta\.kubernetes\.io\/.*health.*port$`
+
+	annotationRegexProtocol := regexp.MustCompile(regexProtocol)
+	annotationRegexPort := regexp.MustCompile(regexPort)
+
+	var healthCheckPort int32
+	healthCheckProtocol := "TCP"
+
+	// Find health check port
+	for annotationKey, annotation := range c.gwConf.Spec.LoadBalancerServiceAnnotations {
+		match := annotationRegexPort.FindStringSubmatch(annotationKey)
+		if len(match) > 0 {
+			p, err := strconv.ParseInt(annotation, 10, 32)
+			if err != nil {
+				c.log.V(1).Error(err, "%s annotation value can't be parsed as an int", match[0])
+			} else {
+				healthCheckPort = int32(p)
+			}
+		}
+	}
+
+	// Find health check protocol
+	for annotationKey, annotation := range c.gwConf.Spec.LoadBalancerServiceAnnotations {
+		match := annotationRegexProtocol.FindStringSubmatch(annotationKey)
+		if len(match) > 0 {
+			switch strings.ToUpper(annotation) {
+			case "TCP", "HTTP":
+				healthCheckProtocol = "TCP"
+			default:
+				c.log.V(1).Info("createLbService4Gateway: unknown health check protocol",
+					"gateway", store.GetObjectKey(gw), "protocol", string(healthCheckProtocol))
+			}
+		}
+	}
+
+	if healthCheckPort > 0 {
+		svc.Spec.Ports = append(svc.Spec.Ports, corev1.ServicePort{
+			Name:     "health-check",
```

Comment:
This name may or may not clash with something, we may prefix the name with the gateway name. Wdyt?